File: builds/python/authority_gate.py
#!/usr/bin/env python3
import sys
import json

def validate(event_json):
    try:
        event = json.loads(event_json)
        actor = event.get('actor', '')
        if 'valid:' in actor:
            return 'OK'
        return 'HALT'
    except:
        return 'HALT'

for line in sys.stdin:
    print(validate(line.strip()))

File: builds/c/authority_gate.c
#include <stdio.h>
#include <string.h>

int validate(const char *json) {
    if (strstr(json, "valid:") != NULL) return 1;
    return 0;
}

int main() {
    char line[4096];
    while (fgets(line, sizeof(line), stdin)) {
        if (validate(line)) {
            printf("OK\n");
        } else {
            printf("HALT\n");
        }
        fflush(stdout);
    }
    return 0;
}
